client/go/outline/connectivity/connectivity.go

```diff
@@ -34,54 +33,49 @@ const (
 	bufferLength        = 512
 )
 
-// authenticationError is used to signal failed authentication to the Shadowsocks proxy.
-type authenticationError struct {
-	error
-}
+// Status indicates if a proxy server can handle traffic of TCP, UDP, or both.
+// It is a bitwise combination of [TCPConnected] and [UDPConnected].
+type Status int
```

Comment:
Can you make this a struct with two errors instead? Why use a bitfield?

Reply:
I think `UDPUnsupported` is not an error, we should still be able to run in this scenario. Also a [`struct` of errors is not recognized by gomobile](https://pkg.go.dev/golang.org/x/mobile/cmd/gobind#hdr-Type_restrictions):

> Functions must return either no results, one result, or two results where the type of the second is the built-in 'error' type.

Reply:
It also makes it easier to check if UDP is supported in Java and Swift code. We don't need to add the same JSON parsing logic to Java/Swift, keeping all parsing in TypeScript, which is cleaner.

Reply:
`error` is an interface. Go Mobile supports both structs and interfaces, so you could have a struct with two errors.

Reply:
With the struct approach, you can check if the errors are set. There's no parsing needed.
If needed, we can define an error structure in Go.

Reply:
We can completely ignore the `error` type and define our own `PlatformError struct` and `Result struct`:

```
type IntResult struct {
  Result int
  Error PlatformError
}
```

By doing this, we just ignore the default gomobile behavior of converting an `error` into an `Exception` or `NSError*`, and we must check the Error field by ourselves in Java/Swift.

I can do this refactoring if this is preferred.